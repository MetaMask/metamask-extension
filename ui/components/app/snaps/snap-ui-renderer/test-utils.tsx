import React from 'react';
import { Reducer } from 'redux';
import { RenderResult } from '@testing-library/react';
import { JSXElement } from '@metamask/snaps-sdk/jsx';
import configureStore, { MetaMaskReduxState } from '../../../../store/store';
import { renderWithProvider } from '../../../../../test/lib/render-helpers';
import mockState from '../../../../../test/data/mock-state.json';
import { SnapUIRenderer } from './snap-ui-renderer';

export const MOCK_SNAP_ID = 'npm:@metamask/test-snap-bip44';
export const MOCK_INTERFACE_ID = 'interfaceId';

type RenderInterfaceOptions = {
  useFooter?: boolean;
  onCancel?: () => void;
  contentBackgroundColor?: string;
  state?: Record<string, unknown>;
};

// The return type from renderWithProvider includes RenderResult plus a history property
type RenderWithProviderResult = RenderResult & {
  history: {
    location: {
      pathname: string;
    };
  };
};

// Combine the renderWithProvider result with our custom properties
type RenderInterfaceResult = RenderWithProviderResult & {
  updateInterface: (
    newContent: JSXElement,
    newState?: Record<string, unknown> | null,
  ) => void;
  getRenderCount: () => number;
};

/**
 * Renders a Snap UI interface for testing purposes.
 *
 * @param content - The JSXElement to render.
 * @param options - The options for rendering the interface.
 * @param options.useFooter - Whether to render the footer.
 * @param options.onCancel - The function to call when the interface is cancelled.
 * @param options.contentBackgroundColor - The background color of the content.
 * @param options.state - The state of the interface.
 * @returns Testing utilities with render result, plus updateInterface and getRenderCount functions.
 */
export function renderInterface(
  content: JSXElement,
  {
    useFooter = false,
    onCancel,
    contentBackgroundColor,
    state = {},
  }: RenderInterfaceOptions = {},
): RenderInterfaceResult {
  const store = configureStore({
    ...mockState,
    metamask: {
      ...mockState.metamask,
      interfaces: {
        [MOCK_INTERFACE_ID]: {
          snapId: MOCK_SNAP_ID,
          content,
          state,
          context: null,
          contentType: null,
        },
      },
    },
  });

  const reducer: Reducer<MetaMaskReduxState> = (reduxState, action) => {
    const storeState = reduxState ?? store.getState();

    if (action.type === 'updateInterface') {
      return {
        ...storeState,
        metamask: {
          ...storeState.metamask,
          interfaces: {
            [MOCK_INTERFACE_ID]: {
              snapId: MOCK_SNAP_ID,
              content: action.content,
              state: action.state ?? reduxState,
              context: null,
              contentType: null,
            },
          },
        },
      };
    }
    return storeState;
  };

  store.replaceReducer(reducer);

  const updateInterface = (
    newContent: JSXElement,
    newState: Record<string, unknown> | null = null,
  ) => {
    store.dispatch({
      type: 'updateInterface',
      content: newContent,
      state: newState,
    });
  };

  const result = renderWithProvider(
    <SnapUIRenderer
      snapId={MOCK_SNAP_ID}
      interfaceId={MOCK_INTERFACE_ID}
      useFooter={useFooter}
      onCancel={onCancel}
      contentBackgroundColor={contentBackgroundColor}
      PERF_DEBUG
    />,
    store,
  );

  const getRenderCount = () =>
    parseInt(
      result.getByTestId('performance').getAttribute('data-renders') as string,
      10,
    );

  return { ...result, updateInterface, getRenderCount };
}
