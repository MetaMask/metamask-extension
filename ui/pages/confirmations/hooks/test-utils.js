import { useSelector } from 'react-redux';

import { useMultichainSelector } from '../../../hooks/useMultichainSelector';

import { GasEstimateTypes } from '../../../../shared/constants/gas';
import { getCurrentCurrency } from '../../../ducks/metamask/metamask';
import {
  getShouldShowFiat,
  txDataSelector,
  getCurrentKeyring,
  getTokenExchangeRates,
  getPreferences,
  selectConversionRateByChainId,
  selectNetworkConfigurationByChainId,
} from '../../../selectors';

import {
  getCustomMaxFeePerGas,
  getCustomMaxPriorityFeePerGas,
} from '../../../ducks/swaps/swaps';
import { Numeric } from '../../../../shared/modules/Numeric';
import { EtherDenomination } from '../../../../shared/constants/common';
import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
import {
  getMultichainCurrentCurrency,
  getMultichainIsEvm,
  getMultichainNativeCurrency,
  getMultichainShouldShowFiat,
} from '../../../selectors/multichain';

// Why this number?
// 20 gwei * 21000 gasLimit = 420,000 gwei
// 420,000 gwei is 0.00042 ETH
// 0.00042 ETH * 100000 = $42
export const MOCK_ETH_USD_CONVERSION_RATE = 100000;

export const LEGACY_GAS_ESTIMATE_RETURN_VALUE = {
  gasEstimateType: GasEstimateTypes.legacy,
  gasFeeEstimates: {
    low: '10',
    medium: '20',
    high: '30',
  },
  estimatedGasFeeTimeBounds: {},
};

export const FEE_MARKET_ESTIMATE_RETURN_VALUE = {
  gasEstimateType: GasEstimateTypes.feeMarket,
  gasFeeEstimates: {
    low: {
      minWaitTimeEstimate: 180000,
      maxWaitTimeEstimate: 300000,
      suggestedMaxPriorityFeePerGas: '3',
      suggestedMaxFeePerGas: '53',
    },
    medium: {
      minWaitTimeEstimate: 15000,
      maxWaitTimeEstimate: 60000,
      suggestedMaxPriorityFeePerGas: '7',
      suggestedMaxFeePerGas: '70',
    },
    high: {
      minWaitTimeEstimate: 0,
      maxWaitTimeEstimate: 15000,
      suggestedMaxPriorityFeePerGas: '10',
      suggestedMaxFeePerGas: '100',
    },
    estimatedBaseFee: '50',
  },
  estimatedGasFeeTimeBounds: {},
};

export const HIGH_FEE_MARKET_ESTIMATE_RETURN_VALUE = {
  gasEstimateType: GasEstimateTypes.feeMarket,
  gasFeeEstimates: {
    low: {
      minWaitTimeEstimate: 180000,
      maxWaitTimeEstimate: 300000,
      suggestedMaxPriorityFeePerGas: '3',
      suggestedMaxFeePerGas: '53000',
    },
    medium: {
      minWaitTimeEstimate: 15000,
      maxWaitTimeEstimate: 60000,
      suggestedMaxPriorityFeePerGas: '7',
      suggestedMaxFeePerGas: '70000',
    },
    high: {
      minWaitTimeEstimate: 0,
      maxWaitTimeEstimate: 15000,
      suggestedMaxPriorityFeePerGas: '10',
      suggestedMaxFeePerGas: '100000',
    },
    estimatedBaseFee: '50000',
  },
  estimatedGasFeeTimeBounds: {},
};

export const generateUseSelectorRouter =
  ({
    checkNetworkAndAccountSupports1559Response,
    shouldShowFiat = true,
  } = {}) =>
  (selector) => {
    if (selector === getMultichainIsEvm) {
      return true;
    }
    if (selector === selectConversionRateByChainId) {
      return MOCK_ETH_USD_CONVERSION_RATE;
    }
    if (selector === getMultichainNativeCurrency) {
      return EtherDenomination.ETH;
    }
    if (selector === getPreferences) {
      return {
        showNativeTokenAsMainBalance: true,
      };
    }
    if (
      selector === getMultichainCurrentCurrency ||
      selector === getCurrentCurrency
    ) {
      return 'USD';
    }
    if (selector === selectNetworkConfigurationByChainId) {
      return '2';
    }
    if (
      selector === getMultichainShouldShowFiat ||
      selector === getShouldShowFiat
    ) {
      return shouldShowFiat;
    }
    if (selector === txDataSelector) {
      return {
        txParams: {
          value: '0x5555',
        },
      };
    }
    if (selector.toString().includes('getTargetAccount')) {
      return {
        balance: '0x440aa47cc2556',
      };
    }
    if (selector === getCustomMaxFeePerGas) {
      return '0x5208';
    }
    if (selector === getCustomMaxPriorityFeePerGas) {
      return '0x5208';
    }
    if (selector.toString().includes('checkNetworkAndAccountSupports1559')) {
      return checkNetworkAndAccountSupports1559Response;
    }
    if (selector === getCurrentKeyring) {
      return { type: '' };
    }
    if (selector === getTokenExchangeRates) {
      return { '0x1': '1' };
    }
    return undefined;
  };

export function convertFromHexToFiat(value) {
  const val = new Numeric(value, 16).toBase(10).toString();
  return `$${(val * MOCK_ETH_USD_CONVERSION_RATE).toFixed(2)}`;
}

export function convertFromHexToETH(value) {
  const val = new Numeric(value, 16, EtherDenomination.WEI)
    .toBase(10)
    .toDenomination(EtherDenomination.ETH);
  return `${val} ETH`;
}

export const configureEIP1559 = () => {
  useGasFeeEstimates.mockImplementation(() => FEE_MARKET_ESTIMATE_RETURN_VALUE);
  useSelector.mockImplementation(
    generateUseSelectorRouter({
      checkNetworkAndAccountSupports1559Response: true,
    }),
  );
  useMultichainSelector.mockImplementation(
    generateUseSelectorRouter({
      checkNetworkAndAccountSupports1559Response: true,
    }),
  );
};

export const configureLegacy = () => {
  useGasFeeEstimates.mockImplementation(() => LEGACY_GAS_ESTIMATE_RETURN_VALUE);
  useSelector.mockImplementation(
    generateUseSelectorRouter({
      checkNetworkAndAccountSupports1559Response: false,
    }),
  );
  useMultichainSelector.mockImplementation(
    generateUseSelectorRouter({
      checkNetworkAndAccountSupports1559Response: false,
    }),
  );
};

export const configure = () => {
  useSelector.mockImplementation(generateUseSelectorRouter());
  useMultichainSelector.mockImplementation(generateUseSelectorRouter());
};
