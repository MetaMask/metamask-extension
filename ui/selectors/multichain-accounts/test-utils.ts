import { AccountGroupId } from '@metamask/account-api';
import { NetworkConfiguration } from '@metamask/network-controller';
import { MultichainNetworkConfiguration } from '@metamask/multichain-network-controller';
import { MultichainNetworkConfigurationsByChainIdState } from '../../../shared/modules/selectors/networks';
import {
  AccountTreeState,
  InternalAccountsState,
  MultichainAccountsState,
} from './account-tree.types';

// Helper function to create properly typed mock state
export const createMockMultichainAccountsState = (
  accountTree: AccountTreeState,
  internalAccounts: InternalAccountsState,
  networkConfigurations?: {
    networkConfigurationsByChainId?: Record<string, NetworkConfiguration>;
    multichainNetworkConfigurationsByChainId?: Record<
      string,
      MultichainNetworkConfiguration
    >;
  },
): MultichainAccountsState & MultichainNetworkConfigurationsByChainIdState => ({
  metamask: {
    accountTree,
    internalAccounts,
    networkConfigurationsByChainId:
      networkConfigurations?.networkConfigurationsByChainId || {},
    multichainNetworkConfigurationsByChainId:
      networkConfigurations?.multichainNetworkConfigurationsByChainId || {},
  },
});

// Helper function to create empty state
export const createEmptyState = (): MultichainAccountsState &
  MultichainNetworkConfigurationsByChainIdState =>
  createMockMultichainAccountsState(
    {
      wallets: {},
      selectedAccountGroup: null as unknown as AccountGroupId,
    },
    {
      accounts: {},
      selectedAccount: '',
    },
  );
