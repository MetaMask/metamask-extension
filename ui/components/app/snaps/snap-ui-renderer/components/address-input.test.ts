import { Box, AddressInput, Field } from '@metamask/snaps-sdk/jsx';
import { fireEvent } from '@testing-library/react';
import { renderInterface } from '../test-utils';

describe('SnapUIAddressInput', () => {
  it('will render', () => {
    const { container, getByRole } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          chainId: 'eip155:0',
        }),
      }),
    );

    const input = getByRole('textbox') as HTMLInputElement;
    expect(input).toBeDefined();
    expect(input.value).toStrictEqual('');

    expect(container).toMatchSnapshot();
  });

  it('supports existing state', () => {
    const { getByRole } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          chainId: 'eip155:0',
        }),
      }),
      {
        state: { input: 'eip155:0:0x1234567890123456789012345678901234567890' },
      },
    );

    const input = getByRole('textbox') as HTMLInputElement;
    expect(input).toBeDefined();
    expect(input.value).toStrictEqual(
      '0x1234567890123456789012345678901234567890',
    );
  });

  it('supports a placeholder', () => {
    const { getByRole } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          placeholder: 'Enter an address',
          chainId: 'eip155:0',
        }),
      }),
    );

    const input = getByRole('textbox') as HTMLInputElement;
    expect(input).toBeDefined();
    expect(input.placeholder).toStrictEqual('Enter an address');
  });

  it('supports the disabled prop', () => {
    const { getByRole } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          disabled: true,
          chainId: 'eip155:0',
        }),
      }),
    );

    const input = getByRole('textbox') as HTMLInputElement;
    expect(input).toBeDefined();
    expect(input).toBeDisabled();
  });

  it('will render within a field', () => {
    const { getByRole, getByText } = renderInterface(
      Box({
        children: [
          Field({
            label: 'Address',
            children: AddressInput({
              name: 'input',
              chainId: 'eip155:0',
            }),
          }),
        ],
      }),
    );

    const input = getByRole('textbox') as HTMLInputElement;
    expect(input).toBeDefined();
    const label = getByText('Address');
    expect(label).toBeDefined();
  });

  it('will render a matched address name', () => {
    const { container, getByRole, getByText } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          chainId: 'eip155:0',
        }),
      }),
    );

    const input = getByRole('textbox') as HTMLInputElement;
    fireEvent.change(input, {
      target: {
        value: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
      },
    });

    const matchedAddressName = getByText('Test Account');
    const matchedAddress = getByText(
      '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
    );
    const avatar = container.querySelector('svg');
    expect(matchedAddressName).toBeDefined();
    expect(matchedAddress).toBeDefined();
    expect(avatar).toBeDefined();

    expect(container).toMatchSnapshot();
  });

  it('will render avatar when displayAvatar is true', () => {
    const testAddress = '0xabcd5d886577d5081b0c52e242ef29e70be3e7bc';

    const { container, getByDisplayValue } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          chainId: 'eip155:0',
          displayAvatar: true,
        }),
      }),
      { state: { input: `eip155:0:${testAddress}` } },
    );

    const matchedAddress = getByDisplayValue(testAddress);
    expect(matchedAddress).toBeDefined();
    const avatar = container.querySelector('svg');
    expect(avatar).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('will not render avatar when displayAvatar is false', () => {
    const testAddress = '0xabcd5d886577d5081b0c52e242ef29e70be3e7bc';

    const { container, getByDisplayValue } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          chainId: 'eip155:0',
          displayAvatar: false,
        }),
      }),
      { state: { input: `eip155:0:${testAddress}` } },
    );

    const matchedAddress = getByDisplayValue(testAddress);
    expect(matchedAddress).toBeDefined();
    const avatar = container.querySelector('svg');
    expect(avatar).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders with an invalid CAIP Account ID', () => {
    const testAddress = 'https://foobar.baz/foobar';

    const { container, getByRole } = renderInterface(
      Box({
        children: AddressInput({
          name: 'input',
          chainId: 'eip155:0',
          displayAvatar: false,
        }),
      }),
      { state: { input: null } },
    );

    const input = getByRole('textbox') as HTMLInputElement;
    fireEvent.change(input, { target: { value: testAddress } });
    expect(container).toMatchSnapshot();
  });
});
