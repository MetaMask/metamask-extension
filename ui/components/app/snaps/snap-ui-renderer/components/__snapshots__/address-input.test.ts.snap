// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapUIAddressInput renders with an invalid CAIP Account ID 1`] = `
<div>
  <div
    class="mm-box snap-ui-renderer__content mm-box--height-full mm-box--background-color-background-alternative"
    style="overflow-y: auto; margin-bottom: 0px;"
  >
    <div
      class="box snap-ui-renderer__container box--display-flex box--flex-direction-column"
    >
      <div
        class="box snap-ui-renderer__panel box--display-flex box--flex-direction-column box--justify-content-flex-start box--color-text-default"
      >
        <div
          class="mm-box mm-form-text-field snap-ui-renderer__address-input mm-box--display-flex mm-box--flex-direction-column"
        >
          <div
            class="mm-box mm-text-field mm-text-field--size-lg mm-text-field--truncate mm-form-text-field__text-field mm-box--padding-right-4 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-lg mm-box--border-width-1 box--border-style-solid"
          >
            <input
              autocomplete="off"
              class="mm-box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md mm-box--margin-0 mm-box--padding-0 mm-box--padding-right-2 mm-box--padding-left-4 mm-box--color-text-default mm-box--background-color-transparent mm-box--border-style-none"
              focused="false"
              id="input"
              type="text"
              value="https://foobar.baz/foobar"
            />
            <span
              class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-info-default"
              style="mask-image: url('./images/icons/close.svg'); cursor: pointer;"
            />
          </div>
        </div>
      </div>
    </div>
    <span
      data-renders="1"
      data-testid="performance"
    />
  </div>
</div>
`;

exports[`SnapUIAddressInput will not render avatar when displayAvatar is false 1`] = `
<div>
  <div
    class="mm-box snap-ui-renderer__content mm-box--height-full mm-box--background-color-background-alternative"
    style="overflow-y: auto; margin-bottom: 0px;"
  >
    <div
      class="box snap-ui-renderer__container box--display-flex box--flex-direction-column"
    >
      <div
        class="box snap-ui-renderer__panel box--display-flex box--flex-direction-column box--justify-content-flex-start box--color-text-default"
      >
        <div
          class="mm-box mm-form-text-field snap-ui-renderer__address-input mm-box--display-flex mm-box--flex-direction-column"
        >
          <div
            class="mm-box mm-text-field mm-text-field--size-lg mm-text-field--truncate mm-form-text-field__text-field mm-box--padding-right-4 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-lg mm-box--border-width-1 box--border-style-solid"
          >
            <input
              autocomplete="off"
              class="mm-box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md mm-box--margin-0 mm-box--padding-0 mm-box--padding-right-2 mm-box--padding-left-4 mm-box--color-text-default mm-box--background-color-transparent mm-box--border-style-none"
              focused="false"
              id="input"
              type="text"
              value="0xabcd5d886577d5081b0c52e242ef29e70be3e7bc"
            />
            <span
              class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-info-default"
              style="mask-image: url('./images/icons/close.svg'); cursor: pointer;"
            />
          </div>
        </div>
      </div>
    </div>
    <span
      data-renders="1"
      data-testid="performance"
    />
  </div>
</div>
`;

exports[`SnapUIAddressInput will render 1`] = `
<div>
  <div
    class="mm-box snap-ui-renderer__content mm-box--height-full mm-box--background-color-background-alternative"
    style="overflow-y: auto; margin-bottom: 0px;"
  >
    <div
      class="box snap-ui-renderer__container box--display-flex box--flex-direction-column"
    >
      <div
        class="box snap-ui-renderer__panel box--display-flex box--flex-direction-column box--justify-content-flex-start box--color-text-default"
      >
        <div
          class="mm-box mm-form-text-field snap-ui-renderer__address-input mm-box--display-flex mm-box--flex-direction-column"
        >
          <div
            class="mm-box mm-text-field mm-text-field--size-lg mm-text-field--truncate mm-form-text-field__text-field mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-lg mm-box--border-width-1 box--border-style-solid"
          >
            <input
              autocomplete="off"
              class="mm-box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md mm-box--margin-0 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--color-text-default mm-box--background-color-transparent mm-box--border-style-none"
              focused="false"
              id="input"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <span
      data-renders="1"
      data-testid="performance"
    />
  </div>
</div>
`;

exports[`SnapUIAddressInput will render a matched address name 1`] = `
<div>
  <div
    class="mm-box snap-ui-renderer__content mm-box--height-full mm-box--background-color-background-alternative"
    style="overflow-y: auto; margin-bottom: 0px;"
  >
    <div
      class="box snap-ui-renderer__container box--display-flex box--flex-direction-column"
    >
      <div
        class="box snap-ui-renderer__panel box--display-flex box--flex-direction-column box--justify-content-flex-start box--color-text-default"
      >
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column"
        >
          <div
            class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--gap-2 mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-lg mm-box--border-color-border-muted mm-box--border-width-1 box--border-style-solid"
            style="height: 48px;"
          >
            <div
              style="display: flex;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 24px; height: 24px; display: inline-block; background: rgb(250, 58, 0);"
              >
                <svg
                  height="24"
                  width="24"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#18CDF2"
                    height="24"
                    transform="translate(-0.786295127845455 -2.478213052095374) rotate(328.9 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#035E56"
                    height="24"
                    transform="translate(-13.723846281624033 7.94434640381145) rotate(176.2 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F26602"
                    height="24"
                    transform="translate(12.500881513667943 -10.653854792247811) rotate(468.9 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
            <div
              class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--align-items-center"
              style="flex: 1; min-width: 0;"
            >
              <div
                class="mm-box mm-box--gap-2 mm-box--flex-direction-column"
                style="min-width: 0; flex: 1;"
              >
                <p
                  class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-box--color-text-default"
                >
                  Test Account
                </p>
                <p
                  class="mm-box mm-text mm-text--body-xs mm-text--ellipsis mm-box--color-text-default"
                >
                  0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc
                </p>
              </div>
            </div>
            <span
              class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-info-default"
              style="mask-image: url('./images/icons/close.svg'); cursor: pointer; flex-shrink: 0;"
            />
          </div>
        </div>
      </div>
    </div>
    <span
      data-renders="1"
      data-testid="performance"
    />
  </div>
</div>
`;

exports[`SnapUIAddressInput will render avatar when displayAvatar is true 1`] = `
<div>
  <div
    class="mm-box snap-ui-renderer__content mm-box--height-full mm-box--background-color-background-alternative"
    style="overflow-y: auto; margin-bottom: 0px;"
  >
    <div
      class="box snap-ui-renderer__container box--display-flex box--flex-direction-column"
    >
      <div
        class="box snap-ui-renderer__panel box--display-flex box--flex-direction-column box--justify-content-flex-start box--color-text-default"
      >
        <div
          class="mm-box mm-form-text-field snap-ui-renderer__address-input mm-box--display-flex mm-box--flex-direction-column"
        >
          <div
            class="mm-box mm-text-field mm-text-field--size-lg mm-text-field--truncate mm-form-text-field__text-field mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--align-items-center mm-box--background-color-background-default mm-box--rounded-lg mm-box--border-width-1 box--border-style-solid"
          >
            <div
              style="display: flex;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 24px; height: 24px; display: inline-block; background: rgb(242, 202, 2);"
              >
                <svg
                  height="24"
                  width="24"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#03405E"
                    height="24"
                    transform="translate(-5.1032274517898335 -4.005457853807921) rotate(369.5 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#FAA200"
                    height="24"
                    transform="translate(4.654017159434154 -10.422671895325516) rotate(387.5 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#016F8E"
                    height="24"
                    transform="translate(-16.380316861028035 -8.91841489898652) rotate(241.5 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
            <input
              autocomplete="off"
              class="mm-box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md mm-box--margin-0 mm-box--padding-0 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--color-text-default mm-box--background-color-transparent mm-box--border-style-none"
              focused="false"
              id="input"
              type="text"
              value="0xabcd5d886577d5081b0c52e242ef29e70be3e7bc"
            />
            <span
              class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-info-default"
              style="mask-image: url('./images/icons/close.svg'); cursor: pointer;"
            />
          </div>
        </div>
      </div>
    </div>
    <span
      data-renders="1"
      data-testid="performance"
    />
  </div>
</div>
`;
